blog/2024-11-linked-signal/README.md

```diff
@@ -0,0 +1,239 @@
+
+```typescript
+@Component({
+  selector: 'app-shopping-cart',
+  template: `
+    <p>Book: {{ selectedBook().title }}</p>
+    <input [(ngModel)]="amount" />
+  `,
+  standalone: true
+})
+class ShoppingCartComponent {
+  selectedBook = input.required<Book>();
+  amount = linkedSignal({
+    source: this.selectedBook,
+    computation: () => 1 // Resets to 1 when selectedBook changes
+  });
+}
+```
+
+In this case, whenever `selectedBook` changes, the `amount` input resets to `1`. 
+This pattern is useful in forms where we want fields to return to a default state when certain inputs change.
+
+## Advanced Scenarios for LinkedSignal
+
+### Nested State Management
+
+Suppose you have nested data such as book properties (`title` and `rating`), and we want these fields to reset when a different `book` is selected. 
+Here’s how we could manage this with `linkedSignal`:
+
+```typescript
+import { Component, input, linkedSignal } from '@angular/core';
+import { Book } from './book';
+
+@Component({
+  selector: 'app-book',
+  template: `
+    <p>Title: {{ title() }}</p>
+    <p>Rating: {{ rating() }}</p>
+
+    <button (click)="doRateUp()">Rate up!</button> 
+  `,
+})
+class BookComponent  {
+  book = input.required<Book>();
+  ratingHasBeenChanged = output<{ isbn: string, newRating: number }>();
+
+  title = linkedSignal({
+    source: this.book,
+    computation: book => book.title,
+  });
+
+  rating = linkedSignal({
+    source: this.book,
+    computation: book => book.rating,
+  });
+
+  doRateUp() {
+    const newRating = this.rating() + 1;
+    this.rating.set(newRating);
+
+    this.ratingHasBeenChanged({ 
+      isbn: this.book().isbn(),
+      newRating
+    });
+  }
+}
+```
+
+With this setup, both `title` and `rating` reset when `book` changes, helping to keep data synchronized in cases where the structure of state is hierarchical or dependent on specific identifiers. 
+While the `linkedSignal` makes sure that the data resets when necessary, we can still update our local state directly. 
+In this example we update `rating` locally and communicate the change back to the parent component.
+
+### Synchronizing Server-Data for Client-Side Edits
+
+`linkedSignal` can also help when dealing with data from a server that we want to edit locally.
+If we're fetching data from an API but need to allow changes on the client side, we can use `linkedSignal` to keep local edits in sync with the original server data.
+Here is an example that uses some books from our API:
+
+
+```typescript
+import { Component, inject, linkedSignal } from '@angular/core';
+import { BookStoreService } from './book-store.service';
+
+@Component({
+  selector: 'app-dashboard',
+  template: `
+    <ul>
+      @for (book of books(); track book.isbn) {
+        <li>{{ book.title }}</li>
+      }
+    </ul>
+
+    <button (click)="sortBooks()">Sort books by rating</button>
```

Comment:
```suggestion
    <button (click)="sortBooks()">Sort books by rating (locally)</button>
```

I had a similar example in the Resource API blogpost and I intentionally called the method `sortBookListLocally` to make this clear.